clap_generate/examples/multicall.rs

```diff
@@ -0,0 +1,18 @@
+use clap::{App, AppSettings};
+use clap_generate::{generate, generators::Bash};
+use std::io;
+
+fn main() {
+    let mut app = App::new(env!("CARGO_CRATE_NAME"))
+        .setting(AppSettings::Multicall)
+        .subcommand(
+            App::new("test")
+                .subcommand(App::new("config"))
+                .subcommand(App::new("generate-completions")),
+        )
+        .subcommand(App::new("hello"));
+    for applet in app.get_subcommands_mut() {
+        let name = applet.get_name().to_owned();
+        generate(Bash, applet, &name, &mut io::stdout());
+    }
+}
```

Comment:
The main question here is are we generating usable completions?

Reply:
Appears to, it worked for my own application, though I limited support to having a subcommand output the completions since I hadn't formed an opinion on how apps installed with `cargo install` should handle completions.
```bash
$ eval "$(../target/debug/examples/multicall)"
$ ln ../target/debug/examples/multicall test
$ ln ../target/debug/examples/multicall hello
$ test <TAB><TAB>
config                generate-completions  -h                    --help                help                  
$ test config -<TAB><TAB>
-h      --help  
$ hello -<TAB><TAB>
-h      --help  
```
I'm not sure how best to demonstrate it works.